cleanlab/outlier.py

```diff
@@ -0,0 +1,534 @@
+# Copyright (C) 2017-2022  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+
+
+import warnings
+import numpy as np
+from cleanlab.count import get_confident_thresholds
+from sklearn.neighbors import NearestNeighbors
+from sklearn.exceptions import NotFittedError
+from typing import Optional, Union, Tuple, Dict
+from cleanlab.internal.label_quality_utils import (
+    _subtract_confident_thresholds,
+    get_normalized_entropy,
+)
+from cleanlab.internal.validation import assert_valid_inputs
+
+
+class OutOfDistribution:
+    """
+    OutOfDistribution = Out of distribution detection for classification examples using intermediate feature
+    embeddings or predicted probabilities. Each passed in example is given an out-of-distribution score. Scores lie in [0,1] with smaller
+    values indicating examples that are less typical under the dataset distribution (values near 0 indicate outliers).
+
+    Parameters
+    ----------
+    params : dict, default = None
+     Optional keyword arguments to that change how estimator is fit. Effect of arguments passed in depends on if
+     `OutOfDistribution` object is fit on `features` or `pred_probs`.
```

Comment:
"Effect of arguments passed in depends on if `OutOfDistribution` object is fit on `features` or `pred_probs`"
->
"Effect of arguments passed in here depend on if `OutOfDistribution` object will be fit to `features` or `pred_probs`"